interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is a mixture of JavaScript and HTML into one essentially. Standard syntax for JS for referencing variables is different. Also, naming conventions for HTML attributes must be in camelCase. The major syntax difference between HTML and JSX tags is that JSX tags have no identifier on them. For example, <h1></h1> are header tags for a webpage and can be identified by the "h1". JSX fragment tags are <></> empty tags without any kind of identifier. The "h1" identifier also always for CSS to be implemented by calling upon the "h1" identifier in a CSS file. JSX does not give that type of implementation. CSS can be written inline or in a separate CSS file and can be referenced inside the JSX tags. 
 
-Researched answer:
+Researched answer: JSX(JavaScript markdown syntax extension): In JSX class names are not used because class is a keyword in JS already. Therefore, in JSX className is the naming convention used to give CSS attributes or behavior that piece of code is doing. 
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package manager that can be installed through a terminal. Yarn is primarily used for testing files and can only be used on test.js extension files. Yarn dependencies must be installed in order to properly run the tests on the provided file. The three folder/files that appear after installation are yarn.lock, node_modules, and package.json. node_modules holds all the available JS libraries. yarn.lock is more of an update log file. package.json allows the dependencies to be installed. 
 
-Researched answer:
+Researched answer: Package.json describes the package. yarn manages all the files/directories and installed dependencies that you are working with. 
 
 3. What is a React component?
 
-Your answer:
+Your answer: A React component is one piece of an entire React application. React components allow for specific, defined functions to reside in. Each React component is independent of one another, but essentially are invoked in a main component. That main component is typically called App.js where in best practice is where all logic for an app resides and component invocation. Data can be passed to components using props. Data can also be passed from a component to the main component using prop functions. React components are typically designed to be reusable for all kinds of data. 
 
-Researched answer:
+Researched answer: React components are mostly made up of JS code and importantly they return markup language to the browser. Components mostly are comprised of functions that become reusable over and over again. 
```

Comment:
This would be a great area to really focus on trying to find something you didn't already know or mention. For example, you could talk about parent and child components.